### PR TITLE
Add construction object size list

### PR DESCRIPTION
**Pull request by brakenium**
I've done some testing with Varunda on the construction objects and we compiled this list. This list shows which construction objects generate which kill experience event and shows if they are small, medium or large. In classic census fashion the Ammo Dispenser is a engineer turret destroy event.

I've used the readme and node script from the weapons folder and adjusted it so it works in here. Since I didn't see much reason in not including columns of this csv file I've decided for it to just sort the original file and then generate a json variant.

I'm curious what you guys think.

### CODE REVIEW

**Comment by fredkilbourn** [comment]
Mentioned in #api-dev but would be good to include item_id field as well in these lists.

**Comment by brakenium** [comment]
Item ID's are now added for each faction and explanation of which ones are chosen in case of multiple has been added to the readme.

**Review by cooltrain7 — APPROVED** [review]
Thanks for going through and finding out about the item ids/group ids. Just for the sake of conventions I'll tag Fred for a check too.

**Comment by fredkilbourn** [comment]
I'd personally like to see format:

`item_id,﻿item,size,experience_id,faction (or faction_id)`

and have multiple entries for each item representing each faction.  Just trying to think about how I'd consume this in a program, I'd almost certainly be mapping on `item_id` directly and in this format that would require remapping all the data or searching in a more expensive way.

If nobody else sees this as an issue I'm fine with it as is.

File: Construction/readme.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+The item ID's are only for the items with item_type_id of 26 and a item_category_id of 139. Some construction objects might be multiple items, like the weapons which might have an item that has the vehicle weapon category specified as well as the infantry ability (139).
+
+Use the raw.githubusercontent.com/*FilePath* if you want to pull from the live files as and when they get updated.
+
+## Instructions
+
+File formats:
+* `size-list.csv` (csv file used as source)
+* `size-list.csv` (generated json file)
```
Comment (carlst99) [inline Construction/readme.md:13]:
```suggestion
* `size-list.json` (generated json file)
```

Them darn copy-paste errors 😛

**Comment by carlst99** [comment]
I agree with @fredkilbourn on his format suggestion. In almost all instances, I'll be working with or receiving the item ID, so making it little easier to index by would be nice.

**Comment by brakenium** [comment]
@fredkilbourn is this what you had in mind? Or should the JSON be indexed by item_id so you would have something like this```
[
        "6005624": {
            "item_name": "Xiphos Anti-Personnel Tower",
            "size": "medium",
            "faction_id": "4",
            "experience_id": "604"
        }
]
```
To anyone: I'm not sure if there should lists with stuff removed, in my opinion we should just keep it easy without too many variants. An example would be the sanction lists where (some) machine lists have columns removed and the none sanction filtered out.

**Comment by fredkilbourn** [comment]
I think having the JSON as a map like the weapons makes sense.  Is instantly in a lookup format but can still be iterated like a list if you like.

I'm fine about the not having lists with missing info thing.

**Comment by brakenium** [comment]
Cleaned up some code and now mapping the JSON on item_id field while removing that field from the object (to only show it once)

**Review by fredkilbourn — APPROVED** [review]
Looks good to me now.

**Pull request merged**